src/measure_theory/lp_space.lean

```diff
@@ -1218,4 +1217,390 @@ by { rw eq, exact continuous_pos_part.comp continuous_neg }
 end pos_part
 end Lp
 end measure_theory
+
 end composition
+
+
+
+/-!
+## `L^p` is a complete space
+
+We show that `L^p` is a complete space for `1 ≤ p`.
+-/
+
+section complete_space
+
+variables [borel_space E] [second_countable_topology E]
+
+namespace measure_theory
+namespace Lp
+
+lemma snorm'_lim_sub {f : ℕ → α → G} {p : ℝ} (hp1 : 1 ≤ p) {f_lim : α → G}
+  (h_lim : ∀ᵐ (x : α) ∂μ, filter.tendsto (λ n, f n x) filter.at_top (𝓝 (f_lim x))) (n : ℕ) :
+  snorm' (f n - f_lim) p μ
+    = (∫⁻ a, filter.at_top.liminf (λ m, (nnnorm (f n a - f m a) : ℝ≥0∞)^p) ∂μ) ^ (1/p) :=
+begin
+  suffices h_no_pow : (∫⁻ a, (nnnorm ((f n - f_lim) a)) ^ p ∂μ)
+    = (∫⁻ a, filter.at_top.liminf (λ (m : ℕ), (nnnorm (f n a - f m a) : ℝ≥0∞)^p) ∂μ),
+  { rw [snorm', h_no_pow], },
+  refine lintegral_congr_ae (h_lim.mono (λ a ha, _)),
+  rw filter.tendsto.liminf_eq,
+  simp_rw [ennreal.coe_rpow_of_nonneg _ (le_trans zero_le_one hp1), ennreal.tendsto_coe],
+  refine ((nnreal.continuous_rpow_const (le_trans zero_le_one hp1)).tendsto
+      (nnnorm (f n a - f_lim a))).comp _,
+  exact (continuous_nnnorm.tendsto (f n a - f_lim a)).comp (tendsto_const_nhds.sub ha),
+end
+
+/-! ### `Lp` is complete iff Cauchy sequences of `ℒp` have limits in `ℒp` -/
+
+lemma tendsto_Lp_of_tendsto_ℒp {ι} [nonempty ι] [linear_order ι] [hp : fact (1 ≤ p)]
+  {f : ι → Lp E p μ} (f_lim : α → E) (f_lim_ℒp : mem_ℒp f_lim p μ)
+  (h_tendsto : ∀ ε, 0 < ε → ε < ∞ → (∃ N, ∀ n, N ≤ n → snorm (f n - f_lim) p μ < ε)) :
```

Comment:
Can't you write this assumption as the fact that `snorm (f n - f_lim) p μ` tends to `0` at infinity, here and in a few lemmas below? When there is a Cauchy-like assumption, I agree it is clearer to phrase it with epsilons, but for straight convergence the tendsto formulation seems clearer.

Reply:
It looked like an easy change... but if I use `ennreal.tendsto_at_top` to translate the hypothesis into epsilons I get <= epsilon, while when I rewrite the goal with `metric.tendsto_at_top` I need to prove < epsilon.
Now I need to go through the work of introducing epsilon/2 (which will likely be tedious mainly because I am working with `ennreal.of_real epsilon`).

I don't know if there is a good reason behind the discrepancy between the two `tendsto_at_top`, but I'd rather have the two lemmas use <= epsilon.

Reply:
I did it and it was faster than expected. I looks like I can then change all the other lemmas to use <= instead of <. There is another place where I introduced epsilon/2, and that should not be necessary anymore.
It will turn into a lot of small changes, but hopefully for the better.